lib/src/Uncertainty/Distribution/StudentCopulaFactory.cxx

```diff
@@ -0,0 +1,130 @@
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/StudentCopulaFactory.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/MaximumLikelihoodFactory.hxx"
+#include "openturns/NormalCopula.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(StudentCopulaFactory)
+
+static const Factory<StudentCopulaFactory> Factory_StudentCopulaFactory;
+
+/* Default constructor */
+StudentCopulaFactory::StudentCopulaFactory()
+  : DistributionFactoryImplementation()
+{
+  // Initialize optimization solver parameter using the ResourceMap
+  String solverName(ResourceMap::GetAsString("StudentCopulaFactory-DefaultOptimizationAlgorithm"));
+  solver_ = OptimizationAlgorithm::Build(solverName);
+  solver_.setMaximumEvaluationNumber(ResourceMap::GetAsUnsignedInteger("StudentCopulaFactory-MaximumEvaluationNumber"));
+  solver_.setMaximumAbsoluteError(ResourceMap::GetAsScalar("StudentCopulaFactory-MaximumAbsoluteError"));
+  solver_.setMaximumRelativeError(ResourceMap::GetAsScalar("StudentCopulaFactory-MaximumRelativeError"));
+  solver_.setMaximumResidualError(ResourceMap::GetAsScalar("StudentCopulaFactory-MaximumObjectiveError"));
+  solver_.setMaximumConstraintError(ResourceMap::GetAsScalar("StudentCopulaFactory-MaximumConstraintError"));
+}
+
+/* Virtual constructor */
+StudentCopulaFactory * StudentCopulaFactory::clone() const
+{
+  return new StudentCopulaFactory(*this);
+}
+
+/* Here is the interface that all derived class must implement */
+
+Distribution StudentCopulaFactory::build(const Sample & sample) const
+{
+  return buildAsStudentCopula(sample).clone();
+}
+
+Distribution StudentCopulaFactory::build(const Point & parameters) const
+{
+  return buildAsStudentCopula(parameters).clone();
+}
+
+Distribution StudentCopulaFactory::build() const
+{
+  return buildAsStudentCopula().clone();
+}
+
+
+StudentCopula StudentCopulaFactory::buildAsStudentCopula(const Sample & sample) const
+{
+  const UnsignedInteger dimension = sample.getDimension();
+  if (sample.getSize() == 0)
+    throw InvalidArgumentException(HERE) << "Error: cannot build a StudentCopula distribution from an empty sample";
+  if (dimension < 2)
+    throw InvalidArgumentException(HERE) << "Error: cannot build a StudentCopula distribution sample of dimension < 2";
```

Comment:
"Error: cannot build a StudentCopula distribution **from a** sample of dimension < 2"